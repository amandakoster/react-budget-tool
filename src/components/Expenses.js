import React, {Component} from 'react';

class Expenses extends Component{
  constructor(props) {
    super(props);

    this.state= {
      budget: 0,
      expenses: [],
    };
    this.handleNewBudget = this.handleNewBudget.bind(this);
  }

  handleNewBudget(e) {
    let budget = e.target.value;
    this.setState({budget});
  }

  render() {

    let totalSpent = 0;
    let totalRemaining = 0;

    return(

      <div id='budgetWrapper'>

        <div>
          <h2>Budget Summary</h2>
          <h3>Budget: {this.state.budget}</h3>
          <h3>Spent: {totalSpent}</h3>
          <h3>Remaining: {totalRemaining}</h3>

          <form>
            <input type="number" 
              onChange={this.handleNewBudget} 
              placeholder="Enter New Budget Amount"></input>
          </form>
        </div>

        <div>
          <h2>Expenses</h2>
          <table>
            <thead>
              <tr>
                <th colSpan='2'>Expense</th>
                <th colSpan='2'>Amount</th>
              </tr>
            </thead>
            <tbody>
              {
                this.state.expenses.map((expenses, i) =>
                  <tr key={expenses.id}>
                    <td><a href="#">x</a></td>
                    <td>{expenses.name}</td>
                    <td>{expenses.amount}</td>
                  </tr>
                    
                )
              }
            </tbody>
          </table>
        </div>

      </div>

    );
  }

}

export default Expenses;
