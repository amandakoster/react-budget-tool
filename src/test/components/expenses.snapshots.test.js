import React from 'react';
import renderer from 'react-test-renderer';

import Expenses from '../../components/Expenses';


describe('<Expenses>', () => {

  test('renders with a clean state', () => {

    let app = {
      state: {
        budget:0,
        expenses: [],
        showErrors: true,
      },
      setState: (data) => app.state = Object.assign( {}, app.state, data),
    };

    const component = renderer.create(
      <Expenses app={app} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });



  test('renders a model when balance <=0', () => {
    let app = {
      state: {
        budget:10,
        expenses: [
          {id:1, name: 'foo', amount: 5 },
          {id:2, name: 'bar', amount: 6 },
        ],
        showErrors:true,
      },
      setState: (data) => app.state = Object.assign({}, app.state, data),
    };

    const component = renderer.create(
      <Expenses app={app} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    
  } );
});