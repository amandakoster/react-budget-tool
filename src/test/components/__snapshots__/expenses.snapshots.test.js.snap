// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Expenses> renders a model when balance <=0 1`] = `
<div
  id="budgetWrapper"
>
  <div>
    <h2>
      Budget Summary
    </h2>
    <h3>
      Budget: 
      10
    </h3>
    <h3>
      Spent: 
      11
    </h3>
    <h3>
      Remaining: 
      -1
    </h3>
    <form
      onSubmit={[Function]}
    >
      <input
        name="budget"
        placeholder="Enter Budget Amount"
        type="number"
      />
      <button>
        Set Budget
      </button>
    </form>
  </div>
  <div>
    <h2>
      Expenses
    </h2>
    <form
      onSubmit={[Function]}
    >
      <input
        name="name"
        onChange={[Function]}
        placeholder="Expense Item"
        type="text"
      />
      <input
        name="amount"
        onChange={[Function]}
        placeholder="Amount"
        type="number"
      />
      <button
        type="submit"
      >
        Add Expense
      </button>
    </form>
    <table>
      <thead>
        <tr>
          <th
            colSpan="2"
          >
            Expense
          </th>
          <th
            colSpan="2"
          >
            Amount
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <a
              data-key={1}
              href="#"
              onClick={[Function]}
            >
              x
            </a>
          </td>
          <td>
            foo
          </td>
          <td>
            5
          </td>
        </tr>
        <tr>
          <td>
            <a
              data-key={2}
              href="#"
              onClick={[Function]}
            >
              x
            </a>
          </td>
          <td>
            bar
          </td>
          <td>
            6
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`<Expenses> renders with a clean state 1`] = `
<div
  id="budgetWrapper"
>
  <div>
    <h2>
      Budget Summary
    </h2>
    <h3>
      Budget: 
      0
    </h3>
    <h3>
      Spent: 
      0
    </h3>
    <h3>
      Remaining: 
      0
    </h3>
    <form
      onSubmit={[Function]}
    >
      <input
        name="budget"
        placeholder="Enter Budget Amount"
        type="number"
      />
      <button>
        Set Budget
      </button>
    </form>
  </div>
  <div>
    <h2>
      Expenses
    </h2>
    <table>
      <thead>
        <tr>
          <th
            colSpan="2"
          >
            Expense
          </th>
          <th
            colSpan="2"
          >
            Amount
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
